src/map/atcommand.c

```diff
@@ -5160,17 +5160,17 @@ ACMD(storeall)
 
 ACMD(clearstorage)
 {
-	int i, j;
+	int i;
 
 	if (sd->state.storage_flag == STORAGE_FLAG_NORMAL) {
 		clif->message(fd, msg_fd(fd,250));
 		return false;
 	}
 
-	j = sd->status.storage.storage_amount;
-	for (i = 0; i < j; ++i) {
-		storage->delitem(sd, i, sd->status.storage.items[i].amount);
+	for (i = 0; i < VECTOR_LENGTH(sd->storage); ++i) {
```

Comment:
As discussed, vector indexes of storage items will be `memset` to 0 while the storage is open. When closed, zeroed entries will be erased from the vector.